### PR TITLE
REF: avoid FutureWarning about using deprecates loc.__setitem__ non-inplace usage

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Not a bugfix AFAICT, the new test passes without this edit.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
could refactor Loc._ensure_iterable_column_indexer to re-use in the get_loc post-processing

File: pandas/core/generic.py [hunk 1]
```diff
@@ -6885,14 +6885,48 @@ def fillna(
...
                         else downcast.get(k)  # type: ignore[union-attr]
                     )
-                    # GH47649
-                    result.loc[:, k] = (
-                        result[k].fillna(v, limit=limit, downcast=downcast_k).values
-                    )
-                    # TODO: result.loc[:, k] = result.loc[:, k].fillna(
-                    #     v, limit=limit, downcast=downcast_k
-                    # )
-                    # Revert when GH45751 is fixed
```
Comment (mroeschke) [inline pandas/core/generic.py:6895]: Is #45751 still applicable to this code path/is it fixed?

**Review by mroeschke — COMMENTED** [review]
Should this be for 1.5 or 2.0?

File: pandas/core/generic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/generic.py:6895]: I don't think so, no

**Comment by jbrockmendel** [comment]
> Should this be for 1.5 or 2.0?

I'd put it into 1.5.x to avoid showing a spurious FutureWarning to users.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**